src/Scripting/CSharpTest/InteractiveSessionTests.cs

```diff
@@ -1776,6 +1775,95 @@ public void HostObjectAssemblyReference3()
             }
         }
 
+        public class E
+        {
+            public bool TryGetValue(out object obj)
+            {
+                obj = new object();
+                return true;
+            }
+        }
+
+        [Fact]
+        [WorkItem(39565, "https://github.com/dotnet/roslyn/issues/39565")]
+        public async Task Regression39565()
+        {
+            var code = @"
+            if(TryGetValue(out var result)){
```

Comment:
Nit: Don't indent the embedded code, it's harder to read and maintain the indentation.

```C#
            var code = @"
if (TryGetValue(out var result)){
   _ = result;
}
return true;
";
```